src/utils/mesh.js

```diff
@@ -242,10 +319,21 @@ async function checkAndInstallMeshPlugin (installedPlugins) {
  * @param installedPlugins
  */
 export async function createMesh (runAIOCommand, installedPlugins) {
-  const { saas, paas, catalog } = config.get('commerce.datasource')
-  const { org: githubOrg, repo: githubRepo } = config.get('commerce.github')
+  const shouldCreateMesh = await confirmAPIMeshCreation()
+
+  if (!shouldCreateMesh) {
+    aioLogger.debug('Not creating API Mesh - will use default environment')
+    return
+  }
+
+  await checkAndInstallMeshPlugin(installedPlugins)
+
+  const { apiKey, environmentId, datasource, github } =
+        config.get('commerce')
+  const { saas, paas, catalog } = datasource
+  const { org: githubOrg, repo: githubRepo } = github
 
-  if (paas || saas) {
+  if ((paas && catalog) || saas) {
```

Comment:
catalog is not necessary a requirement for PaaS - you can have just PaaS and roll your own catalog service. That's why I left it out.